client/src/languages/rPython.ts

```diff
@@ -0,0 +1,327 @@
+import {h} from 'maquette';
+import * as Langs from '../definitions/languages'; 
+import * as Graph from '../definitions/graph'; 
+import * as Values from '../definitions/values'; 
+import {printPreview} from '../editors/preview'; 
+import {createEditor} from '../editors/editor';
+import {Md5} from 'ts-md5';
+import axios from 'axios';
+
+// declare var PYTHONSERVICE_URI: string;
+// declare var RSERVICE_URI: string;
+declare var DATASTORE_URI: string;
+
+// ------------------------------------------------------------------------------------------------
+// rPython plugin
+// ------------------------------------------------------------------------------------------------
+
+/// A class that represents a Python block. All blocks need to have 
+/// `language` and Python also keeps the Python source we edit and render
+
+type languageType = "python" | "r"
+
+export class rPythonBlockKind implements Langs.Block {
```

Comment:
Again, I think this should not be just `rPythonBlockKind`. We need a better name.

 - Maybe `LanguageServiceBlockKind` (but that's a bit long)
 - or `ApiBlockKind` (but that's not very clear)
 - or maybe `ExternalBlockKind` ?

If you have better ideas, please share them! I think I like `ExternalBlockKind` the most out of the above.

Reply:
Yes, Nick and I were talking about it. We weren't happy with rPythonPlugin either. We thought apiBasedLanguagePlugin, but then Gamma is api based isn't it. I'll keep this open for more suggestions.